### DISCUSSION TITLE
Looking to make adjustments to login and pop-up dialogs (CSS contributions)

### DISCUSSION THREAD

**Discussion by Chris-Behling** [dnnsoftware/Dnn.Platform#4268] [General]
Hey there, I've been getting the itch to look into making CSS contributions to the platform and I'm not entirely sure where to start. There are a few areas that I'm really interested in seeing improvements in but I'm not sure how these areas may be changing as we approach v10/v11.

I'd love to see both the default login and pop-up modal/dialogs refined to look and feel like an extension of the persona bar and more inline with the current version of DNN as opposed to DNN 7/8.

Most of these styles seem to be associated to the default.css file (/Resources/Shared/stylesheets/dnndefault/7.0.0/default.css). Snooping through the repo I'm not seeing any scss/sass for these files and was trying to identify if they are compiled or currently managed through pure css. If it is the case that these styles are managed through standard CSS maybe that would be a starting place, to shift these files to scss.

**Comment by bdukes** [comment]
Those files are just CSS, without any compilation.  The Persona Bar uses Less, so if you wanted to look into adding a build process to the `Website` files, I think that would be the preferred language.

If you're going to make significant changes to the display of the front-end that don't require users to opt-in, we would probably target those changes for v10 (assuming everyone agrees that they're changes we want in the platform).  I think @david-poindexter would be the first line of approval for anything like that.

**Reply by bdukes** [reply]
Regarding the build process piece, I'd be happy to talk through what it might look like to set that up (@valadas and @donker may also have thoughts)

**Reply by Chris-Behling** [reply]
Hey there, I'd be happy to discuss this more and see how I could get involved. As I'm not entirely sure what the roadmap for DNN looks like it's hard to know where I can jump in and start helping out. As @david-poindexter pointed out, I definitely understand the concern for modifying markup as it can introduce breaking changes. Pure CSS add-ons could be implemented but large scale changes even from a CSS perspective could cause unintended breaking changes.

A v11 update may even be more appropriate if the goal for v10 is to have as few breaking changes while removing telerik dependencies.

I'd think updating the css to use a processor like less could be done fairly easily and would be a good idea.

Is the plan for v11 to even keep /dnndefault/7.0.0/default.css files? From my perspective I'd love to see the old UI removed from components like the Module Settings, and Login. I'm feeling that all of DNN should shift to using the same styles found in the Persona bar for consistency.

**Reply by bdukes** [reply]
I believe @valadas is working on a new trimmed down `default.css` for v10, and hoping to incorporate components which would let us replace the old DNN UX in a number of places.

**Comment by david-poindexter** [comment]
I'd be happy work with you @10PoundGorilla-ChrisBehling.  

Many people have tried to improve the login control styling, etc., but it usually comes back to how and when.  Getting consensus on the "how" has been historically challenging.  ;-)  Once the "how" has been decided, that usually dictates the "when".  We follow semantic versioning.  Since many commercial and custom themes already style the login controls, any changes to DNN Platform in this area would be a "breaking change".  So...such changes would need to be planned accordingly.
